src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightsService.java

```diff
@@ -462,4 +500,13 @@ public QueryInsightsHealthStats getHealthStats() {
             topQueriesHealthStatsMap
         );
     }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    private void deleteExpiredIndices() {
+        for (MetricType metricType : MetricType.allMetricTypes()) {
+            topQueriesServices.get(metricType).deleteExpiredIndices(clusterService.state().metadata().indices());
```

Comment:
for `clusterService.state().metadata().indices()`: I feel this is a little bit dangerous. we are assuming users won't ever have indices named `top_queries-*`. Otherwise we will accidentally delete user data.

I think we should at least add another hash string in the index name, maybe `top_queries-<hash>-YYYY.MM.dd` ? The hash should be computed using a constant so that we can keep the pattern consistent after the OpenSearch process restarts.